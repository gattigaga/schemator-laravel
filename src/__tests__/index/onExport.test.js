import { onExport } from "../../index";

Date.now = jest.fn(() => 1534939718031);

describe("onExport()", () => {
  it("should returns exported result", () => {
    const project = {
      pluginID: "gattigaga-laravel",
      name: "myproject",
      zoom: 100,
      timestamp: Date.now()
    };

    const tables = [
      {
        id: expect.any(String),
        name: "User",
        options: [
          {
            id: "id",
            label: "ID",
            isChecked: true
          },
          {
            id: "rememberToken",
            label: "Remember Token",
            isChecked: true
          },
          {
            id: "softDeletes",
            label: "Soft Deletes",
            isChecked: false
          },
          {
            id: "timestamps",
            label: "Timestamps",
            isChecked: true
          }
        ],
        position: { x: 128, y: 128 },
        timestamp: Date.now()
      },
      {
        id: expect.any(String),
        name: "Post",
        options: [
          {
            id: "id",
            label: "ID",
            isChecked: true
          },
          {
            id: "rememberToken",
            label: "Remember Token",
            isChecked: false
          },
          {
            id: "softDeletes",
            label: "Soft Deletes",
            isChecked: false
          },
          {
            id: "timestamps",
            label: "Timestamps",
            isChecked: true
          }
        ],
        position: { x: 320, y: 320 },
        timestamp: Date.now()
      }
    ];

    const userFields = [
      {
        id: expect.any(String),
        tableID: tables[0].id,
        name: "name",
        type: "string"
      },
      {
        id: expect.any(String),
        tableID: tables[0].id,
        name: "email",
        type: "string"
      },
      {
        id: expect.any(String),
        tableID: tables[0].id,
        name: "password",
        type: "text"
      }
    ];

    const postFields = [
      {
        id: expect.any(String),
        tableID: tables[1].id,
        name: "user_id",
        type: "integer"
      },
      {
        id: expect.any(String),
        tableID: tables[1].id,
        name: "title",
        type: "string"
      }
    ];

    const fields = [...userFields, ...postFields];

    const relations = [
      {
        id: expect.any(String),
        fieldID: postFields[0].id,
        fromTableID: postFields[0].tableID,
        toTableID: tables[0].id
      }
    ];

    const destinationPath = "/exported";

    const data = {
      project,
      tables,
      fields,
      relations
    };

    const result = onExport(destinationPath, data);

    expect(result).toMatchSnapshot();
  });
});
