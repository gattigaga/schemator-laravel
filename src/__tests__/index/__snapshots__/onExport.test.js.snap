// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onExport() should returns exported result 1`] = `
Object {
  "files": Array [
    Object {
      "content": "<?php

namespace App;

use Illuminate\\\\Database\\\\Eloquent\\\\Model;

class User extends Model
{
    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'name',
        'email',
        'password'
    ];
}",
      "path": "/exported/myproject/app/User.php",
    },
    Object {
      "content": "<?php

namespace App;

use Illuminate\\\\Database\\\\Eloquent\\\\Model;

class Post extends Model
{
    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'user_id',
        'title'
    ];
}",
      "path": "/exported/myproject/app/Post.php",
    },
    Object {
      "content": "<?php

use Illuminate\\\\Support\\\\Facades\\\\Schema;
use Illuminate\\\\Database\\\\Schema\\\\Blueprint;
use Illuminate\\\\Database\\\\Migrations\\\\Migration;

class CreateUsersTable extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        Schema::create('users', function (Blueprint $table) {
            $table->increments('id');
            $table->string('name');
            $table->string('email');
            $table->text('password');
            $table->rememberToken();
            $table->timestamps();
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        Schema::dropIfExists('users');
    }
}",
      "path": "/exported/myproject/database/migrations/2018_08_22_190838_create_users_table.php",
    },
    Object {
      "content": "<?php

use Illuminate\\\\Support\\\\Facades\\\\Schema;
use Illuminate\\\\Database\\\\Schema\\\\Blueprint;
use Illuminate\\\\Database\\\\Migrations\\\\Migration;

class CreatePostsTable extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        Schema::create('posts', function (Blueprint $table) {
            $table->increments('id');
            $table->integer('user_id')->unsigned();
            $table->foreign('user_id')->references('id')->on('users');
            $table->string('title');
            $table->timestamps();
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        Schema::dropIfExists('posts');
    }
}",
      "path": "/exported/myproject/database/migrations/2018_08_22_190838_create_posts_table.php",
    },
  ],
  "paths": Array [
    "/exported/myproject",
    "/exported/myproject/app",
    "/exported/myproject/database",
    "/exported/myproject/database/migrations",
  ],
}
`;
